src/Fixer/ClassNotation/FinalPublicMethodForAbstractClassFixer.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class FinalPublicMethodForAbstractClassFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $magicMethods = [
```

Comment:
this is a near copy of the list here `https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.13.1/src/Fixer/Casing/MagicMethodCasingFixer.php#L26` , maybe we can move this list to the `PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer` ?

Reply:
I would be :+1: for that  , but lets hear from others as well

Reply:
could you do this change @Slamdunk ? after that this PR seems RTM to me (I can do the move later if you can't, just let me know :) )

Reply:
For this PR `FunctionsAnalyzer` could be useful, but for [MagicMethodCasingFixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.13.1/src/Fixer/Casing/MagicMethodCasingFixer.php#L26)? That fixer must have the full map anyway because it needs a casing map that for `FunctionsAnalyzer` would be useless

Reply:
I think both fixers can build the map they need based on a list of magic method names, so we only need to maintain that list at one place?

Reply:
@keradus is this discussion blocking the merge of the PR?